File: src/frontend/src/types/Product.ts
export type ProductStatus = 'Draft' | 'Active' | 'Archived'

export interface Product {
  id: string
  name: string
  sku: string
  price: number
  status: ProductStatus
  categoryId: string
  brandId: string | null
  createdAt: string
  updatedAt: string
}

export interface CreateProductRequest {
  name: string
  sku: string
  price: number
  categoryId: string
  brandId?: string | null
  status?: ProductStatus
}

export interface UpdateProductRequest {
  name: string
  price: number
  categoryId: string
  brandId?: string | null
  status: ProductStatus
}

File: src/backend/Models/Product.cs
namespace DevFlow.Api.Models;

public class Product
{
    public Guid Id { get; set; }
    public string Name { get; set; } = string.Empty;
    public string Sku { get; set; } = string.Empty;
    public decimal Price { get; set; }
    public ProductStatus Status { get; set; } = ProductStatus.Draft;
    public DateTime CreatedAt { get; set; }
    public DateTime UpdatedAt { get; set; }

    public Guid CategoryId { get; set; }
    public Category Category { get; set; } = null!;

    public Guid? BrandId { get; set; }
    public Brand? Brand { get; set; }
}
